src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborWriter.Map.cs

```diff
@@ -38,14 +41,199 @@ public void WriteEndMap()
                 EnsureWriteCapacity(1);
                 _buffer[_offset++] = CborInitialByte.IndefiniteLengthBreakByte;
             }
+
+            AdvanceDataItemCounters();
         }
 
         public void WriteStartMapIndefiniteLength()
         {
             EnsureWriteCapacity(1);
             WriteInitialByte(new CborInitialByte(CborMajorType.Map, CborAdditionalInfo.IndefiniteLength));
-            AdvanceDataItemCounters();
             PushDataItem(CborMajorType.Map, expectedNestedItems: null);
+            _currentKeyOffset = _offset;
+            _currentValueOffset = null;
+        }
+
+        //
+        // Map encoding conformance
+        //
+
+        private bool ConformanceRequiresUniqueKeys()
+        {
+            return ConformanceLevel switch
+            {
+                CborConformanceLevel.Rfc7049Canonical => true,
+                CborConformanceLevel.Ctap2Canonical => true,
+                CborConformanceLevel.Strict => true,
+                CborConformanceLevel.NonStrict => false,
+                _ => false,
```

Comment:
I find it hard to imagine a conformance mode other than the default lax mode that doesn't have this as a requirement.

`return ConformanceLevel != CborConformanceLevel.NonStrict;` is OK.  A classic switch is probably easier on the eyes here; and if you're using either kind of switch consider adding an assert in the default case that reports the enum value that wasn't mapped in.